src/platform/test/functional/page_objects/unified_tabs.ts

```diff
@@ -37,10 +64,100 @@ export class UnifiedTabsPageObject extends FtrService {
     await this.testSubjects.click('unifiedTabs_tabsBar_newTabBtn');
     await this.retry.waitFor('the new tab to appear', async () => {
       const newNumberOfTabs = await this.getNumberOfTabs();
-      return newNumberOfTabs === numberOfTabs + 1;
+      return (
+        newNumberOfTabs === numberOfTabs + 1 &&
+        (await this.getSelectedTab())?.index === newNumberOfTabs - 1
+      );
+    });
+  }
+
+  public async selectTab(index: number) {
+    const tabElements = await this.getTabElements();
+    if (index < 0 || index >= tabElements.length) {
+      throw new Error(`Tab index ${index} is out of bounds`);
+    }
+    await tabElements[index].click();
+    await this.retry.waitFor('the selected tab to change', async () => {
+      return (await this.getSelectedTab())?.index === index;
     });
   }
 
+  public async closeTab(index: number) {
+    const tabElements = await this.getTabElements();
+    if (index < 0 || index >= tabElements.length) {
+      throw new Error(`Tab index ${index} is out of bounds`);
+    }
+    const closeButton = await tabElements[index].findByCssSelector(
+      '[data-test-subj^="unifiedTabs_closeTabBtn_"]'
+    );
+    await closeButton.click();
+    await this.retry.waitFor('the tab to be closed', async () => {
+      return (await this.getNumberOfTabs()) === tabElements.length - 1;
+    });
+  }
+
+  public async openTabMenu(index: number) {
+    const tabElements = await this.getTabElements();
+    if (index < 0 || index >= tabElements.length) {
+      throw new Error(`Tab index ${index} is out of bounds`);
+    }
+    const menuButton = await tabElements[index].findByCssSelector(
+      '[data-test-subj^="unifiedTabs_tabMenuBtn_"]'
+    );
+    await menuButton.click();
+    await this.retry.waitFor('the menu to open', async () => {
+      return (await this.getContextMenuItems()).length > 0;
+    });
```

Comment:
maybe there is something more reliable to track when context menu loading is finished?

Reply:
Context menu does not load anything. It only renders the items. That's what I am checking here.